File: Testing/Code for Testing/Early Basic Syntax Testing/bitfield/test_bitfield.js
// bit field: ready = 1 bits
// bit field: fault = 1 bits
// bit field: mode = 3 bits
// bit field: priority = 3 bits
function createStatusFlags() {
    return { ready: 0, fault: 0, mode: 0, priority: 0 };
}


function main() {
    let flags = createStatusFlags();
    flags.ready = 1;
    flags.fault = 0;
    flags.mode = 5;
    flags.priority = 3;
    console.log(flags.ready);
    console.log(flags.fault);
    console.log(flags.mode);
    console.log(flags.priority);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();

File: Testing/Code for Testing/Early Basic Syntax Testing/bitfield/test_bitfield.py
from dataclasses import dataclass, field

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)

@dataclass
class StatusFlags:
    ready: int = 0  # bit field: 1 bits
    fault: int = 0  # bit field: 1 bits
    mode: int = 0  # bit field: 3 bits
    priority: int = 0  # bit field: 3 bits


def main():
    flags = StatusFlags()
    flags.ready = 1
    flags.fault = 0
    flags.mode = 5
    flags.priority = 3
    print(flags.ready)
    print(flags.fault)
    print(flags.mode)
    print(flags.priority)
    return 0


if __name__ == "__main__":
    main()
